src/content/it/fundamentals/performance/rendering/optimize-javascript-execution.md

```diff
@@ -0,0 +1,192 @@
+    var nextTask = taskList.pop();
+
+    // Process nextTask.
+    processTask(nextTask);
+
+    // Go again if there’s enough time to do the next task.
+    taskFinishTime = window.performance.now();
+  } while (taskFinishTime - taskStartTime < 3);
+
+  if (taskList.length > 0)
+    requestAnimationFrame(processTaskList);
+
+}
+```
+
+Ci sono conseguenze UX e UI per questo approccio e dovrai assicurarti che
+l'utente sappia che un compito è in fase di elaborazione, [usando un indicatore
+di progresso o di
+attività](https://www.google.com/design/spec/components/progress-activity.html)
+. In ogni caso questo approccio manterrà libero il thread principale della tua
+app, aiutandolo a rimanere reattivo alle interazioni dell'utente.
+
+## Conosci la "frame tax" del tuo JavaScript
+
+Quando si valuta un framework, una libreria o il proprio codice, è importante
+valutare quanto costa eseguire il codice JavaScript su una base frame-by-frame.
+Ciò è particolarmente importante quando si eseguono operazioni di animazione
+critiche per le prestazioni come la transizione o lo scorrimento.
+
+Il pannello delle prestazioni di Chrome DevTools è il modo migliore per misurare
+il costo del tuo JavaScript. Normalmente ottieni record di basso livello come
+questo:
+
+<img src="images/optimize-javascript-execution/low-js-detail.png" alt="A
+performance recording in Chrome DevTools">
+
+La sezione **Main** fornisce una flame chart delle chiamate JavaScript in modo
+da poter analizzare esattamente quali funzioni sono state chiamate e quanto
+tempo è stato speso per ognuna.
+
+Grazie a queste informazioni è possibile valutare l'impatto delle prestazioni
+JavaScript sulla propria applicazione e iniziare a trovare e correggere
+eventuali hotspot in cui le funzioni impiegano troppo tempo per essere eseguite.
+Come accennato in precedenza dovresti cercare di rimuovere JavaScript di
+esecuzione lunga o, se ciò non è possibile, spostarlo su un Web Worker liberando
+il thread principale per continuare con altre attività.
+
+Per informazioni su come utilizzare il pannello delle prestazioni consulta
+[Guida introduttiva all'analisi delle prestazioni
+runtime](/web/tools/chrome-devtools/evaluate-performance/) .
+
+## Evita la micro-ottimizzazione del tuo JavaScript
+
+Può essere interessante sapere che il browser può eseguire una versione di
+un'attività 100 volte più velocemente di un'altra cosa, come una richiesta e gli
+elementi `offsetTop` è più veloce del calcolo `getBoundingClientRect()` , ma è
+quasi sempre userai queste funzioni un piccolo numero di volte per frame, quindi
+è normalmente sprecato lo sforzo di concentrarsi su questo aspetto delle
+prestazioni JavaScript. In genere risparmi solo frazioni di millisecondi.
+
+Se stai realizzando un gioco o un'applicazione onerosa dal punto di vista
+computazionale probabilmente sei un'eccezione a questa guida, dato che di solito
+stai calcolando un sacco di calcoli in un singolo frame, e in tal caso tutto
```

Comment:
si cerca di includere tutti i possibili calcoli in ogni singolo frame, e in tal caso tutto